ckb-bin/src/subcommand/run.rs

```diff
@@ -180,6 +182,35 @@ fn verify_genesis(shared: &Shared) -> Result<(), ExitCode> {
         })
 }
 
+fn check_spec(shared: &Shared, args: &RunArgs) -> Result<(), ExitCode> {
+    let store = shared.store();
+    if let Some(spec_hash) = store.get_chain_spec_hash() {
+        if args.chain_spec_hash != spec_hash && !args.skip_chian_spec_check {
+            eprintln!(
+                "chain_spec_hash mismatch Config({}) storage({})",
```

Comment:
```suggestion
                "chain_spec_hash mismatch Config({}) storage({}), pass command line argument --skip-spec-check if you are sure that the two different chains are compatible.",
```

The suggested change may require `cargo fmt`